test/test_jit.py

```diff
@@ -2714,6 +2714,30 @@ def t2(a):
 
         self.checkScript(t2, (torch.zeros(1, 1, 2)))
 
+    def test_addmm_grad(self):
+        class LinearTest(torch.jit.ScriptModule):
+            def __init__(self, param_init):
+                super(LinearTest, self).__init__()
+                self.param = torch.nn.Parameter(param_init)
+
+            @torch.jit.script_method
+            def forward(self, x):
+                return torch.addmm(x, x, self.param)
+
+        param_init = torch.rand(2, 2)
+        lt = LinearTest(param_init)
+        x = torch.rand(2, 2)
+        y = lt(x)
+        y.sum().backward()
+        grad_test = lt.param.grad
```

Comment:
A nicer fix really would be to make `checkScript` check the correctness of gradients (analogously to `checkTrace`).